### PR TITLE
trivial: add missing rpc help messages, remove segwit references, dashify help text, undashify code comments

### PR DESCRIPTION
**Pull request by kwvg**
## Issue being fixed or feature implemented
This pull request is a follow-up to [some](https://github.com/dashpay/dash/pull/5834#discussion_r1470105685) [feedback](https://github.com/dashpay/dash/pull/5834#discussion_r1467009815) received on [dash#5834](https://github.com/dashpay/dash/pull/5834) as the patterns highlighted were present in different parts of the codebase and hence not corrected within the PR itself but addressed separately.

This is that separate PR 🙂 (with some additional cleanup of my own)

## What was done?
* This pull request will remain a draft until [dash#5834](https://github.com/dashpay/dash/pull/5834) as it will introduce more changes that will need to be corrected in this PR.
* Code introduced that is unique to Dash Core (CoinJoin, InstantSend, etc.) has been excluded from un-Dashification as the purpose of it is to reduce backport conflicts, which don't apply in those cases.
  * `CWallet::CreateTransaction` and the `CreateTransactionTest` fixture have been excluded as the former originates from [dash#3668](https://github.com/dashpay/dash/pull/3668) and the latter from [dash#3667](https://github.com/dashpay/dash/pull/3667) and are distinct enough to be unique to Dash Core.
* There are certain Dashifications and SegWit-removals that prove frustrating as it would break compatibility with programs that rely on the naming of certain keys
  * `getrawmempool`, `getmempoolancestors`, `getmempooldescendants` and `getmempoolentry` return `vsize` which is currently an alias of `size`. I have been advised to retain `vsize` in lieu of potential future developments. (this was originally remedied in 219a1d08973e7ccda6e778218b9a8218b4aae034 but has since been dropped)
  * `getaddressmempool`, `getaddressutxos` and `getaddressdeltas` all return a value with the key `satoshis`. This is frustrating to rename to `duffs` for compatibility reasons.
  * `decodepsbt` returns (if applicable) `non_witness_utxo` which is frustrating to rename simply to `utxo` for the same reason.
  * `analyzepsbt` returns (if applicable) `estimated_vsize` which frustrating to rename to `estimated_size` for the same reason.

## Breaking Changes
None

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by knst — APPROVED** [review]
utACK

Seems as there's no breaking changes in this PR.

Btw, PR description mentions some possible changes such as rename "non_witness_utxo" which annoys me too - let's prepare one more PR for v21?

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge. No breaking changes detected. @thephez please confirm style is correct

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -369,7 +369,7 @@ static UniValue gettxchainlocks(const JSONRPCRequest& request)
                 {RPCResult::Type::OBJ, "", "",
                 {
                     {RPCResult::Type::NUM, "height", "The block height"},
-                    {RPCResult::Type::BOOL, "chainlock", "Chainlock status for the block containing the transaction"},
+                    {RPCResult::Type::BOOL, "chainlock", "The state of the corresponding block ChainLock"},
```
Comment (thephez) [inline src/rpc/rawtransaction.cpp:372]: Nit: imo the original wording was easier to understand

**Review by thephez — COMMENTED** [review]
Looks okay to me. Just one nit, but nothing that prevents merging.

File: src/rpc/rawtransaction.cpp (hunk 1 above)
Comment (kwvg) [reply src/rpc/rawtransaction.cpp:372]: The phrase "The state of the corresponding block ChainLock" was [used](https://github.com/dashpay/dash/blob/8ea9c406558c14aab9deff57c6209e506c8f74d1/src/wallet/rpcwallet.cpp#L1405) [more](https://github.com/dashpay/dash/blob/8ea9c406558c14aab9deff57c6209e506c8f74d1/src/rpc/rawtransaction.cpp#L190) and seemed more concise.

**Pull request merged**